### DISCUSSION TITLE
Cloudflare/Workerd support

### DISCUSSION THREAD

**Discussion by bjankord** [auth0/node-oauth2-jwt-bearer#180] [Ideas]
### Checklist

- [X] I have looked into the [Readme](https://github.com/auth0/node-oauth2-jwt-bearer/tree/main/packages/express-oauth2-jwt-bearer#readme) and [Examples](https://github.com/auth0/node-oauth2-jwt-bearer/blob/main/packages/express-oauth2-jwt-bearer/EXAMPLES.md), and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/node-oauth2-jwt-bearer/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Describe the problem you'd like to have solved

I've been working through https://developer.auth0.com/resources/code-samples/spa/vue/basic-authentication and then the https://developer.auth0.com/resources/code-samples/api/express/basic-authorization guides and was hoping I could take the logic used in the Express app example and migrate it over to either a Cloudflare Worker or Pages app.

I was optimistic that might be possible, but knew that most likely, the express-oauth2-jwt-bearer package was not going to work in Cloudflares runtime environment (workerd). Sure enough, soon as I got to that point, I realized it wasn't going to work due to compat issues between Node.js and Cloudflare's Workerd runtimes.

Are there any plans to make a Cloudflare compatible alternative to the express-oauth2-jwt-bearer package or maybe there already is one I haven't found yet. I'm looking for any resources that could help me achieve the equivalent of the  https://developer.auth0.com/resources/code-samples/api/express/basic-authorization tutorial except using Cloudflare Workers or Pages.

### Describe the ideal solution

The ideal solution, in my opinion, is that Auth0 would provide a Cloudflare based oauth2-jwt-bearer package. That way I could pull in the `auth` function from that package and not have to worry about compat issues from a package that was designed specifically for Node/Express with the express-oauth2-jwt-bearer package

**Comment by arpit-jn** [comment]
Hi @bjankord,

Thanks for your interest in using Auth0 with Cloudflare Workers. I think this would be better suited as a discussion topic rather than an issue, since the current `express-oauth2-jwt-bearer` library is specifically built for Express.js on Node.js environments. Supporting Workers would require significant architectural changes.

This seems to be better suited for a discussion hence I'm moving this to gauge interest and explore potential solutions.

For now, please consider these options for Cloudflare Workers:

- Use [jose](https://github.com/panva/jose) directly for JWT validation.
- Check out [Auth0-CF-Worker-Template](https://github.com/auth0-samples/auth0-cf-worker-template).
- Implement custom verification specific to your Worker.

Thanks for bringing this up. Looking forward to continuing the conversation in the Discussions section.
